alright, the failure I get on linux it seems is
```
.cc.o
onnx> [ 97%] Building CXX object CMakeFiles/onnx_gtests.dir/onnx/test/cpp/test_main.cc.o
onnx> [ 98%] Linking CXX executable onnx_gtests
onnx> /nix/store/rhhll3vwpj38ri72ahrrrvcbkhz4fhh6-binutils-2.40/bin/ld: CMakeFiles/onnx_
gtests.dir/onnx/test/cpp/data_propagation_test.cc.o: in function `onnx::Test::DataPropag
ationImplTest_SymbolicShapeTest_Test::TestBody()':
onnx> data_propagation_test.cc:(.text+0x310f): undefined reference to `google::protobuf:
:internal::ArenaStringPtr::Set(absl::lts_20230125::string_view, google::protobuf::Arena*
)'
onnx> /nix/store/rhhll3vwpj38ri72ahrrrvcbkhz4fhh6-binutils-2.40/bin/ld: CMakeFiles/onnx_
gtests.dir/onnx/test/cpp/function_context_test.cc.o: in function `onnx::Test::FunctionAP
ITest_ContextDependentFunctionTest_Test::TestBody()':
onnx> function_context_test.cc:(.text+0x36fb): undefined reference to `google::protobuf:
:internal::ArenaStringPtr::Set(absl::lts_20230125::string_view, google::protobuf::Arena*
)'
onnx> /nix/store/rhhll3vwpj38ri72ahrrrvcbkhz4fhh6-binutils-2.40/bin/ld: CMakeFiles/onnx_
gtests.dir/onnx/test/cpp/function_context_test.cc.o: in function `onnx::Test::FunctionAP
ITest_TypeContextTest_Test::TestBody()':
onnx> function_context_test.cc:(.text+0x48bc): undefined reference to `google::protobuf:
:internal::ArenaStringPtr::Set(absl::lts_20230125::string_view, google::protobuf::Arena*
)'
onnx> /nix/store/rhhll3vwpj38ri72ahrrrvcbkhz4fhh6-binutils-2.40/bin/ld: CMakeFiles/onnx_
onnx> function_verify_test.cc:(.text+0x4e26): undefined reference to `google::protobuf::'
onnx> /nix/store/rhhll3vwpj38ri72ahrrrvcbkhz4fhh6-binutils-2.40/bin/ld: CMakeFiles/onnx_gtests.dir/onnx/test/cpp/function_verify_test.cc.o: in function `onnx::Test::RegisterFunctionSchema()':
onnx> function_verify_test.cc:(.text+0x5f42): undefined reference to `google::protobuf::internal::ArenaStringPtr::Set(absl::lts_20230125::string_view, google::protobuf::Arena*)'
onnx> /nix/store/rhhll3vwpj38ri72ahrrrvcbkhz4fhh6-binutils-2.40/bin/ld: CMakeFiles/onnx_gtests.dir/onnx/test/cpp/function_verify_test.cc.o:function_verify_test.cc:(.text+0x5f7f): more undefined references to `google::protobuf::internal::ArenaStringPtr::Set(absl::lts_20230125::string_view, google::protobuf::Arena*)' follow
onnx> collect2: error: ld returned 1 exit status
onnx> make[2]: *** [CMakeFiles/onnx_gtests.dir/build.make:246: onnx_gtests] Error 1
onnx> make[1]: *** [CMakeFiles/Makefile2:252: CMakeFiles/onnx_gtests.dir/all] Error 2
onnx> make[1]: *** Waiting for unfinished jobs....
```

--- a/pkgs/development/python-modules/onnx/default.nix
+++ b/pkgs/development/python-modules/onnx/default.nix
@@ -13,6 +13,7 @@
 , pythonOlder
 , tabulate
 , typing-extensions
+, abseil-cpp
 }:
 
 let
@@ -34,6 +35,10 @@ in buildPythonPackage rec {
   nativeBuildInputs = [
     cmake
     pybind11
+  ];
+
+  buildInputs = [
+    abseil-cpp
   ];
 
   propagatedBuildInputs = [
@@ -61,6 +66,10 @@ in buildPythonPackage rec {
       --replace 'include(googletest)' ""
     substituteInPlace cmake/unittest.cmake \
       --replace 'googletest)' ')'
+
+    # remove this override in 1.15 that will enable to set the CMAKE_CXX_STANDARD with cmakeFlags
+    substituteInPlace CMakeLists.txt \
+      --replace 'CMAKE_CXX_STANDARD 11' 'CMAKE_CXX_STANDARD 17'
   '';
 
   preConfigure = ''

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -7510,7 +7510,12 @@ self: super: with self; {
   onlykey-solo-python = callPackage ../development/python-modules/onlykey-solo-python { };
 
   onnx = callPackage ../development/python-modules/onnx {
-    protobuf = protobuf3;
+    # in linux c++ defaults to 17
+    # on darwin clang is on an old version so c++ defaults to 11
+    abseil-cpp = if stdenv.isLinux then
+        pkgs.abseil-cpp_202301
+      else
+        pkgs.abseil-cpp_202301.override { cxxStandard = "17"; };
   };
 
   onnxconverter-common = callPackage ../development/python-modules/onnxconverter-common {

